MuteSpotifyAds/SpotifyManager.swift

```diff
@@ -49,10 +65,19 @@ class SpotifyManager: NSObject {
             }
         })
         
-        DispatchQueue.global(qos: .default).async {
-            self.startSpotify(foreground: true)
-            
-            _ = self.trackChanged()
+        if startSpotifyATS {
+            DispatchQueue.global(qos: .default).async {
+                self.startSpotify(foreground: true)
+                _ = self.trackChanged()
+            }
+        } else {
+            var app: String = ""
+            DispatchQueue.global(qos: .default).async {
+                while !app.contains("Applications/Spotify") {
```

Comment:
A infinity while loop is often very performance inefficient. In this case the app checks for Spotify 24/7 without any delay, which is not good in my opinion.

Instead we should listen for some event that gets triggered when Spotify starts. A easy way is to listen for file change events. You can see which files Spotify accesses at start with the following command (requires `sudo`):

```
sudo echo ; /Applications/Spotify.app/Contents/MacOS/Spotify& ; SPOTIFY_PID=$! ; sudo opensnoop -p $SPOTIFY_PID
```

We can observe that Spotify accesses for example `recently_played.bnk.tmp`, which we already observe to know when a track changes. Therefore we can just leave the while out, we don't need it, since track changed gets triggered by the file observer.

Reply:
Therefore, you can remove the whole else statement here.

Reply:
Thank you for your review! I knew it was a very inefficient solution but I couldn't find a better solution.